### PR TITLE
Add Watchlist Method

### PR DESCRIPTION
**Pull request by scottusrobus**
This PR adds an interface for our customers watchlist endpoint, where api users can do the following for a given customer:

1. retrieve a watchlist 
2. add items from a watchlist
3. remove items from a watchlist

See example below for the proposed interface for the PHP Client.

```
# list watchlist items
\VHX\Watchlist::items(array(
  'customer' => $customer
));

# add a watchlist item
\VHX\Watchlist::addItem(array(
   'customer' => $customer,
   'video' => $video
));

# remove a watchlist item
\VHX\Watchlist::removeItem(array(
   'customer' => $customer,
   'video' => $video
));
```

cc @ksheurs

### CODE REVIEW

File: lib/Videos.php [hunk 1]
```diff
@@ -22,6 +22,6 @@ public static function update($id = null, $params = array()) {
 
   // deprecated (same as files)
   public static function allFiles($id = null, $params = array()) {
-    return self::_files($id, $params);
+    return self::_items($id, $params, 'files');
```
Comment (ksheurs) [inline lib/Videos.php:25]: Nice!

**Comment by ksheurs** [comment]
Looks great, Scott! Nice we're re-using that `_items` interface. It pains me our own client libraries don't use `href` :)

**Comment by scottusrobus** [comment]
thanks, yeah thought that made sense for an optimization!

re: `href` the client libraries support passing either hrefs or ids. so we are covered there. that's what this method https://github.com/vhx/vhx-php/blob/master/lib/Resource.php#L40 is for.

is that what you mean? @ksheurs

**Comment by scottusrobus** [comment]
@ksheurs both these will work. 

```
# list watchlist items (via customer href)
\VHX\Watchlist::items(array(
  'customer' => 'https://api.vhx.tv/customers/1'
));

# list watchlist items (via customer id)
\VHX\Watchlist::items(array(
  'customer' => 1
));
```

**Pull request merged**